openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1004 @@
+      + table.panelOptions.withDescription('Details for the agents for OpenStack Neutron.')
+      + table.standardOptions.withOverridesMixin([
+        table.fieldOverride.byName.new('Admin state')
+        + table.fieldOverride.byName.withProperty('custom.displayMode', 'color-text')
+        + table.fieldOverride.byName.withPropertiesFromOptions(
+          table.standardOptions.withMappings(
+            {
+              type: 'value',
+              options: {
+                down: {
+                  color: 'red',
+                  index: 1,
+                  text: 'Down',
+                },
+                up: {
+                  color: 'green',
+                  index: 0,
+                  text: 'Up',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+            },
+            includeByName: {},
+            indexByName: {
+              Time: 6,
+              Value: 9,
+              __name__: 7,
+              adminState: 3,
+              hostname: 5,
+              id: 4,
+              instance: 1,
+              job: 8,
+              service: 0,
+              zone: 2,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              id: 'ID',
+              instance: 'Instance',
+              service: 'Service',
+              zone: 'Zone',
+            },
+          },
+        },
+      ]),
+
+    volumes:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volumes',
+        targets=[t.volumes],
+        description='The number of volumes managed by Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeStatus:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volume status',
+        targets=[t.volumeErrorStatus, t.volumeNonErrorStatus],
+        description='The current status of volumes in Cinder.',
+      )
+      + timeSeries.options.legend.withPlacement('right')
+      + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+        group: 'A',
+        mode: 'normal',
+      })
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Volume usage',
+        targets=[t.volumeUsage],
+        description='The percent of volume storage in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    backupUsage:
```

Comment:
Legend looks fairly strange here as well.
<img width="794" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/5abc4684-371f-4a78-8a48-bdbe8727273c">

Reply:
If the SUO gets restarted the same things will show up as being different. If you don't zoom out so far in time then there should just be a single set (see screenshots).

Reply:
Any reason why the legend got moved to the bottom? Intentional?